src/TropicalGeometry/variety_binomial.jl

```diff
@@ -0,0 +1,42 @@
+################################################################################
+#
+#  Tropicalization of binomial ideals
+#
+#  WARNING: assumes without test that `gens(I)` is binomial
+#
+################################################################################
+
+function tropical_variety_binomial(I::MPolyIdeal,nu::TropicalSemiringMap; weighted_polyhedral_complex_only::Bool=false)
+    ###
+    # Construct matrix of exponent vector differences
+    # and vector of coefficient valuation differences
+    ###
+    G = gens(I)
+    A = matrix(ZZ,[first(collect(expv))-last(collect(expv)) for expv in exponents.(G)])
+    b = [ QQ(nu(last(collect(coeff))/first(collect(coeff)))) for coeff in coefficients.(G)]
+
+    ###
+    # Compute tropical variety multiplicity
+    ###
+    snfAdiag = elementary_divisors(A)
+    weight = abs(prod([m for m in snfAdiag if !iszero(m)]))
+
+    ###
+    # Constructing tropical variety set-theoretically
+    ###
+    A = QQMatrix(A)
```

Comment:
```suggestion
    A = matrix(QQ, A)
```
Note that this automatically makes your code type unstable, at least with the current Julia which doesn't deal well with a variable switching from one type to another. This can be solved by using a new variable name for the new matrix.